File: frontend/src/app/models/Commande.ts
import { CommandLigne } from "./CommandLigne"
import { Statut } from "./Statut"
import { Client } from "./client"
import { User } from "./user"

export class Commande{
commandeId:number
commandeNumero:string
commandeDate:Date
commandeTotalHt:number
commandeTotalTtc:number
commandeTotalTaxes:number
commandesNbrArticles:number
commandeRevenue:number=0
client:Client
clientId:number
clientIntitule:string
clientCode:string
commandeDateLivraisonPrevue:Date
dateCreate:Date
userCreate:User
statut:Statut
statutId:number
lignes:CommandLigne[]
constructor(commandeId:number,commandeNumero:string,commandeDate:Date,commandeTotalHt:number
    ,commandeTotalTtc:number,commandeTotalTaxes:number,commandesNbrArticles:number,client:Client,
    clientId:number,clientIntitule:string,clientCode:string,commandeDateLivraisonPrevue:Date,dateCreate:Date,userCreate:User,statut:Statut
    ,statutId:number,lignes:CommandLigne[]){
this.commandeId=commandeId
this.commandeNumero=commandeNumero
this.commandeDate=commandeDate
this.commandeTotalHt=commandeTotalHt
this.commandeTotalTtc=commandeTotalTtc
this.commandeTotalTaxes=commandeTotalTaxes
this.commandesNbrArticles=commandesNbrArticles
this.client=client
this.clientId=clientId
this.clientIntitule=clientIntitule
this.clientCode=clientCode
this.commandeDateLivraisonPrevue=commandeDateLivraisonPrevue
this.dateCreate=dateCreate
this.userCreate=userCreate
this.statut=statut
this.statutId=statutId
this.lignes=lignes

}
}

File: backend/B2BApplication/Models/Commande.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace B2BApplication.Models
{
    public class Commande
    {
        [Key]
        public int commandeId { get; set; }
        public string commandeNumero { get; set; }
        public DateTime? commandeDate { get; set; } = DateTime.Now;
        public decimal commandeTotalHt { get; set; }
        public decimal commandeTotalTtc { get; set; }
        public decimal commandeTotalTaxes { get; set; }
        public decimal commandeRevenue { get; set; }
        public int commandesNbrArticles { get; set; }
        public Client client { get; set; }
        [ForeignKey("client")]
        public int clientId { get; set; }
        public string clientIntitule { get; set; }
        public string clientCode { get; set; }
        //added by admin when validate
        public DateTime? commandeDateLivraisonPrevue { get; set; }
        public DateTime dateCreate { get; set; }=DateTime.Now;
        public User userCreate { get; set; }
        [ForeignKey("userCreate")]
        public string userCreateId { get; set; }
        public Statut statut { get; set; }
        [ForeignKey("statut")]
        public int statutId { get; set; }
        public ICollection<CommandeLigne> lignes { get; set; }
        public DemandeRetour demandeRetour { get; set; }


    }
}
